tests/test_density_baselines.py

```diff
@@ -0,0 +1,86 @@
+"""Testing simple density estimation baselines for IRL."""
+
+import numpy as np
+import pytest
+
+from imitation import util
+from imitation.density_baselines import (STATE_ACTION_DENSITY, STATE_DENSITY,
+                                         STATE_STATE_DENSITY, DensityReward,
+                                         DensityTrainer)
+
+
+def parametrize_density_stationary(fn):
+  decorator = pytest.mark.parametrize("density_type,is_stationary",
+                                      [(STATE_DENSITY, True),
+                                       (STATE_DENSITY, False),
+                                       (STATE_ACTION_DENSITY, True),
+                                       (STATE_STATE_DENSITY, True)])
+  return decorator(fn)
+
+
+def score_trajectories(trajectories, reward_fn):
+  # score trajectories under given reward function w/o discount
+  returns = []
+  for traj in trajectories:
+    traj_zip = zip(traj['obs'], traj['act'], traj['obs'][1:])
+    ret = 0.0
+    for step, (obs, act, next_obs) in enumerate(traj_zip):
+      ret += reward_fn([obs], [act], [next_obs], steps=[step])
+    returns.append(ret)
+  return np.mean(returns)
+
+
+@parametrize_density_stationary
+def test_density_reward(density_type, is_stationary):
+  # use Pendulum b/c I don't handle termination correctly yet
```

Comment:
This refers to #40: like AIRL/GAIL, this density baselines don't correctly handle episodes that terminate before reaching the horizon. I'm using Pendulum instead of Cartpole because Pendulum doesn't terminate early. I'll update the comment.